File: gpt-3.5-finetuned/stage2/loop_construct_collapse_clause.c
#include "acc_testsuite.h"
#ifndef T1
//T1:loop construct collapse clause,V:2.7-3.3
int test1(){
	int err = 0;
	srand(SEED);
	int sep = 10;
	real_t * a = (real_t *)malloc(10 * n * sizeof(real_t));
	real_t * b = (real_t *)malloc(10 * n * sizeof(real_t));
	real_t * c = (real_t *)malloc(n * sizeof(real_t));
	real_t * d = (real_t *)malloc(n * sizeof(real_t));
	real_t temp = 0;

	for (int x = 0; x < 10 * n; ++x){
		a[x] = rand() / (real_t)(RAND_MAX / 10);
		b[x] = rand() / (real_t)(RAND_MAX / 10);
	}
	for (int x = 0; x < 10; ++x){
		c[x] = rand() / (real_t)(RAND_MAX / 10);
		d[x] = 0.0;
	}

	#pragma acc data copyin(a[0:10*n], b[0:10*n]) copyin(c[0:10]) copyout(d[0:10])
	{
		#pragma acc parallel loop collapse(2) private(temp)
		for (int x = 0; x < 10; ++x){
			temp = 0;
			for (int y = 0; y < n; ++y){
				temp += a[x * n + y] + b[x * n + y];
			}
			d[x] = temp;
		}
	}

	for (int x = 0; x < 10; ++x){
		for (int y = 0; y < n; ++y){
			if (fabs(d[x]) > PRECISION){
				err += 1;
			}
		}
	}

	return err;
}
#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: gpt-3.5-finetuned/stage2/loop_construct_collapse_clause.f90
!T1:loop construct collapse clause,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"

        INTEGER, PARAMETER :: LOOP_COUNT = 2
        INTEGER, PARAMETER :: N = 10
        REAL(8) :: a(LOOP_COUNT, LOOP_COUNT, N)
        REAL(8) :: b(LOOP_COUNT, LOOP_COUNT, N)
        REAL(8) :: c(LOOP_COUNT, LOOP_COUNT)

        INTEGER :: x, y, z
        INTEGER :: errors = 0

        SEEDDIM(1) = 1
        SEEDDIM(1) = SEED
        CALL RANDOM_SEED(PUT=SEEDDIM)

        DO x = 1, LOOP_COUNT
          DO y = 1, LOOP_COUNT
            CALL RANDOM_NUMBER(a(x, y, :))
            b(x, y, :) = 0
          END DO
        END DO

        !$acc data copyin(a(1:LOOP_COUNT,1:LOOP_COUNT,1:N)) copyout(c(1:LOOP_COUNT,1:LOOP_COUNT), b(1:LOOP_COUNT,1:LOOP_COUNT,1:N))
          !$acc parallel
            !$acc loop collapse(LOOP_COUNT)
            DO x = 1, LOOP_COUNT
              !$acc loop collapse(LOOP_COUNT)
              DO y = 1, LOOP_COUNT
                !$acc loop
                DO z = 1, N
                  b(x, y, z) = a(x, y, z)
                END DO
                c(x, y) = 0
                !$acc loop.REDUCTION(+:c)
                DO z = 1, N
                  c(x, y) = c(x, y) + b(x, y, z)
                END DO
              END DO
            END DO
          !$acc end parallel
        !$acc end data

        DO x = 1, LOOP_COUNT
          DO y = 1, LOOP_COUNT
            DO z = 1, N
              IF (b(x, y, z) .ne. a(x, y, z)) THEN
                errors = errors + 1
              END IF
            END DO

            IF (c(x, y) .ne. SUM(b(x, y, 1:N))) THEN
              errors = errors + 1
            END IF
          END DO
        END DO

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END

      PROGRAM main
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"

        INTEGER :: failcode, testrun
        LOGICAL :: failed
        LOGICAL :: test1
        failed = .FALSE.
        failcode = 0
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
        CALL EXIT (failcode)
      END PROGRAM